modules/xfeatures2d/src/latch.cpp

```diff
@@ -502,15 +503,27 @@ namespace cv
 
 
             Mat grayImage;
-            GaussianBlur(image, grayImage, cv::Size(3, 3), 2, 2);
-
-            if (image.type() != CV_8U) cvtColor(image, grayImage, COLOR_BGR2GRAY);
```

Comment:
BTW, Problem is here: `GaussianBlur()` result was erased by `cvtColor()` for non-gray images.